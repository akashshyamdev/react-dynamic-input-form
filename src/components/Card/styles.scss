.card {
    background: #ffffff;
    border: 1px solid rgb(235, 235, 235);
    width: 250px;
    padding: 10px;

    &__header {
        height: 160px;

        img {
            height: 100%;
            width: 100%;
            background-size: cover;
        }
    }

    &__description {
        font-size: 18px;
        font-family: Arial, Helvetica, sans-serif;
        margin: 10px 0px;
        text-align: justify;
    }

    &__action {
        display: flex;
        justify-content: flex-end;

        button {
            border: none;
            background: #065fd4;
            color: #fff;
            height: 35px;
            padding: 0px 20px;
            border-radius: 3px;
            cursor: pointer;
            font-family: Arial, Helvetica, sans-serif;
            text-transform: uppercase;
            font-weight: bold;
        }
    }
    
}