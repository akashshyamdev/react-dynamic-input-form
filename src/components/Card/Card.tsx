import * as React from 'react';

import './styles.css'
import './styles.scss'

import { CardProps } from './../../interfaces'

const Card: React.FC<CardProps> = (props: CardProps) => {
    return (
        <div className="card">
            <header className="card__header">
                <img src={props.photo} />
            </header>
            <main className="card__description">
                {props.text}
            </main>
            <footer className="card__action">
                <button
                    onClick={(event: React.MouseEvent) => {props.handleClick(event)}}>
                        read
                    </button>
            </footer>
        </div>
    );
};

export default Card;
