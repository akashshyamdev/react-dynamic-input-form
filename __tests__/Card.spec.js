import React from "react";
import { create } from "react-test-renderer";
import Card from './../src/components/Card'
import photo from './../src/assets/photo.jpg'

describe("Button component", () => {
    test("Matches the snapshot", () => {
        const button = create(<Card
            text="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam vel odio libero. Nulla arcu enim, ultrices at nibh sed, iaculis efficitur urna. Donec et felis a nunc porta cursus. Duis accumsan nunc diam, sed suscipit libero sodales ut."
            photo={photo}
            handleClick={() => alert('Hello')}
        />);
        expect(button.toJSON()).toMatchSnapshot();
    });
    test("it shows the expected text when clicked (testing the wrong way!)", () => {
        const component = create(<Card
            text="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam vel odio libero. Nulla arcu enim, ultrices at nibh sed, iaculis efficitur urna. Donec et felis a nunc porta cursus. Duis accumsan nunc diam, sed suscipit libero sodales ut."
            photo={photo}
            handleClick={() => alert('Hello')}
        />);
        const instance = component.root;
        expect(instance.props.text).toBe("Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam vel odio libero. Nulla arcu enim, ultrices at nibh sed, iaculis efficitur urna. Donec et felis a nunc porta cursus. Duis accumsan nunc diam, sed suscipit libero sodales ut.");
      });
});


