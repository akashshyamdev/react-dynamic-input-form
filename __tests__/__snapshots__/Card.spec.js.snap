// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches the snapshot 1`] = `
<div
  className="card"
>
  <header
    className="card__header"
  >
    <img
      src="photo.jpg"
    />
  </header>
  <main
    className="card__description"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam vel odio libero. Nulla arcu enim, ultrices at nibh sed, iaculis efficitur urna. Donec et felis a nunc porta cursus. Duis accumsan nunc diam, sed suscipit libero sodales ut.
  </main>
  <footer
    className="card__action"
  >
    <button
      onClick={[Function]}
    >
      read
    </button>
  </footer>
</div>
`;
